src/plasma/plasma_manager.cc

```diff
@@ -642,19 +647,16 @@ int read_object_chunk(ClientConnection *conn, PlasmaRequestBuffer *buf) {
   }
   r = read(conn->fd, buf->data + conn->cursor, s);
 
-  if (r == -1) {
-    LOG_ERROR("read error");
-  } else if (r == 0) {
-    LOG_DEBUG("end of file");
+  if (r <= 0) {
+    LOG_ERROR("Read error");
+    return errno;
   } else {
```

Comment:
Given that we return in the `if` block, I have a slight preference for removing this `else` block.